ftp/src/main/scala/akka/stream/alpakka/ftp/impl/FtpIOGraphStage.scala

```diff
@@ -114,3 +121,70 @@ private[ftp] trait FtpIOGraphStage[FtpClient, S <: RemoteFileSettings]
   }
 
 }
+
+private[ftp] trait FtpIOSinkStage[FtpClient, S <: RemoteFileSettings]
+    extends FtpIOGraphStage[FtpClient, S, SinkShape[ByteString]] {
+  val shape: SinkShape[ByteString] = SinkShape(Inlet[ByteString](s"$name.in"))
+  val in: Inlet[ByteString] = shape.inlets.head.asInstanceOf[Inlet[ByteString]]
+
+  def createLogicAndMaterializedValue(inheritedAttributes: Attributes) = {
+
+    val matValuePromise = Promise[IOResult]()
+
+    val logic = new FtpGraphStageLogic[ByteString, FtpClient, S](shape, ftpLike, connectionSettings, ftpClient) {
+
+      private[this] var osOpt: Option[OutputStream] = None
+      private[this] var writtenBytesTotal: Long = 0L
+
+      setHandler(in,
+        new InHandler {
+        override def onPush(): Unit = {
+          write(grab(in))
+          pull(in)
+        }
+
+        override def onUpstreamFinish(): Unit =
+          try {
+            osOpt.foreach(_.close())
+            disconnect()
+          } finally {
+            matSuccess()
+            super.onUpstreamFinish()
+          }
+      }) // end of handler
+
+      override def postStop(): Unit =
+        try {
+          osOpt.foreach(_.close())
+        } finally {
+          super.postStop()
+        }
+
+      protected[this] def doPreStart(): Unit = {
+        val tryOs = ftpLike.storeFileStream(path.toAbsolutePath.toString, handler.get)
+        if (tryOs.isSuccess) {
+          osOpt = tryOs.toOption
+          pull(in)
+        } else
+          tryOs.failed.foreach { case NonFatal(t) => throw t }
+      }
```

Comment:
This just mimics regular exception handling in a more complicated way. What about just calling `get` on the returned `Try` of `storeFileStream`, it will throw the exception if there is one and return the `OutputStream` if there is one.